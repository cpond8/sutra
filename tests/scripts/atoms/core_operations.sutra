;; Core atom operation tests
;; Tests all implemented atom operations without unimplemented features

;; Test path-based operations
(print "Testing path operations...")
(set! test.value 42)
(print (get test.value))
(del! test.value)
(print (get test.value))

;; Test binary operations with validation
(print "Testing binary operations with validation...")
(print (/ 10 2))
(print (mod 5 2))
(print (mod 7 3))

;; Test unary boolean operations
(print "Testing unary boolean operations...")
(print (not true))
(print (not false))

;; Test binary numeric operations
(print "Testing binary numeric operations...")
(print (eq? 1 1))
(print (eq? 1 2))
(print (gt? 3 2))
(print (lt? 1 2))
(print (gte? 2 2))
(print (lte? 1 2))
(print (- 5 2))

;; Test n-ary numeric operations
(print "Testing n-ary numeric operations...")
(print (+ 1 2 3))
(print (* 2 3 4))

;; Test basic print operations
(print "Testing print operations...")
(print "direct print test")
(print 42)
(print true)
(print false)
