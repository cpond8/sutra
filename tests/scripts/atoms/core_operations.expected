Testing path operations...
42
nil
Testing binary operations with validation...
5
1
1
Testing unary boolean operations...
false
true
Testing binary numeric operations...
true
false
true
true
true
true
3
Testing n-ary numeric operations...
6
24
Testing print operations...
direct print test
42
true
false
